freqtrade/rpc/api_server.py

```diff
@@ -0,0 +1,156 @@
+        app.add_url_rule('/start', 'start', view_func=self.start, methods=['GET'])
+        app.add_url_rule('/daily', 'daily', view_func=self.daily, methods=['GET'])
+
+    def run(self):
+        """ Method that runs flask app in its own thread forever """
+
+        """
+        Section to handle configuration and running of the Rest server
+        also to check and warn if not bound to a loopback, warn on security risk.
+        """
+        rest_ip = self._config['api_server']['listen_ip_address']
+        rest_port = self._config['api_server']['listen_port']
+
+        logger.info('Starting HTTP Server at {}:{}'.format(rest_ip, rest_port))
+        if not IPv4Address(rest_ip).is_loopback:
+            logger.info("SECURITY WARNING - Local Rest Server listening to external connections")
+            logger.info("SECURITY WARNING - This is insecure please set to your loopback,"
+                        "e.g 127.0.0.1 in config.json")
+
+        # Run the Server
+        logger.info('Starting Local Rest Server')
+        try:
+            app.run(host=rest_ip, port=rest_port)
+        except Exception:
+            logger.exception("Api server failed to start, exception message is:")
+
+    def cleanup(self) -> None:
+        pass
+
+    def send_msg(self, msg: Dict[str, str]) -> None:
+        pass
+
+    """
+    Define the application methods here, called by app.add_url_rule
+    each Telegram command should have a like local substitute
+    """
+
+    def page_not_found(self, error):
+        """
+        Return "404 not found", 404.
+        """
+        return json.dumps({
+            'status': 'error',
+            'reason': '''There's no API call for %s''' % request.base_url,
+            'code': 404
+        }), 404
+
+    def hello(self):
+        """
+        None critical but helpful default index page.
+
+        That lists URLs added to the flask server.
+        This may be deprecated at any time.
+        :return: index.html
+        """
+        rest_cmds = 'Commands implemented: <br>' \
+                    '<a href=/daily?timescale=7>/daily?timescale=7</a>' \
+                    '<br>' \
+                    '<a href=/stop>/stop</a>' \
+                    '<br>' \
+                    '<a href=/start>/start</a>'
+        return rest_cmds
+
+    def daily(self):
+        """
+        Returns the last X days trading stats summary.
+
+        :return: stats
+        """
+        try:
+            timescale = request.args.get('timescale')
+            logger.info("LocalRPC - Daily Command Called")
+            timescale = int(timescale)
+
+            stats = self._rpc_daily_profit(timescale,
+                                           self._config['stake_currency'],
+                                           self._config['fiat_display_currency']
+                                           )
+
+            stats = dumps(stats, indent=4, sort_keys=True, default=str)
```

Comment:
Not sure about using `json.dumps` ... 

consider using `flask.jsonify` - it sets the return Mimetype to `application/json`  (and a few other things ...)
[documentation](http://flask.pocoo.org/docs/1.0/api/#flask.json.jsonify)

Reply:
@xmatthias You're right, the `Content-Type` header is wrong with json.dumps. It wasn't my intention to push this, I've experimented a little bit with different frameworks locally (aiohttp, tornado, flask) and for simplicity reasons I changed it to json.dumps so I don't have to touch as many lines while experimenting

Reply:
What is the feeling on jinja templates for the output?

This kind of pushes the client down the HTML route, but with console and local browsers available this might not be a bad choice - over as example a curses front-end

Reply:
For this case jinja templates are overkill, because we just want json responses which are effectively python dicts, so that we are as flexible as possible. However templates could be used in the web-ui to render proper html, but I'm not sure if it makes more sense to implement thte web-ui in react, angular or something like else

Reply:
@gcarq that's fine - as long as we're aware about it :+1: 
which of the ones you tried worked best then?

I personally wouldn't include the web-ui into the main repository, but create a separate one (probably under the freqtrade org) ...

Reply:
jsonify has problems with arrow date strings from the db
json.dumps will serialize without problem - went back to dumps.

Reply:
let's outwait the decision about the framework (see above). If it's flask, it should be jsonify as it sets the header correctly (or that needs to be taken care manually - if the response is json, then the header should be set accordingly - some clients have problems / require manual steps otherwise).

Fixing the time serialization issue (it's not about Arrow - but about the date-time format) is easy in flask - but requires to overwrite/subclass JSON Encoder (https://stackoverflow.com/questions/43663552/keep-a-datetime-date-in-yyyy-mm-dd-format-when-using-flasks-jsonify)